{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyNpmd20+OT/flGitAU/JXVO",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/marcusvdl/healthier/blob/main/healthier_v1.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# DOWNLOADING DATA"
      ],
      "metadata": {
        "id": "mVWX1bNFqPgT"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#!/usr/local/bin/python3\n",
        "\n",
        "\"\"\"---------------------------------------------------------------------------\n",
        "A utility script for downloading data from the Harmonizome, with the ability\n",
        "to configure which datasets and which download types from which to download.\n",
        "Note that all content decompressed is roughly 30GB. The default is to not\n",
        "decompress the files on download.\n",
        "Dependencies:\n",
        "- requests is an HTTP library with an easy-to-use API:\n",
        "  http://docs.python-requests.org/en/latest/\n",
        "---------------------------------------------------------------------------\"\"\"\n",
        "\n",
        "import requests\n",
        "import os\n",
        "import zlib\n",
        "\n",
        "def _download_file(response, filename):\n",
        "    with open(filename, 'wb') as f:\n",
        "        for chunk in response.iter_content(chunk_size=1024):\n",
        "            f.write(chunk)\n",
        "\n",
        "def _download_and_decompress_file(response, filename):\n",
        "    decompressor = zlib.decompressobj(16 + zlib.MAX_WBITS)\n",
        "    filename = filename[:-3]\n",
        "    with open(filename, 'w+') as f:\n",
        "        while True:\n",
        "            chunk = response.raw.read(1024)\n",
        "            if not chunk:\n",
        "                break\n",
        "            string = decompressor.decompress(chunk)\n",
        "            f.write(string)\n",
        "\n",
        "#decompress = True or False\n",
        "def download_datasets(selected_datasets, selected_downloads, decompress=False):\n",
        "    for dataset, path in selected_datasets:\n",
        "        if not os.path.exists(dataset):\n",
        "            os.mkdir(dataset)\n",
        "\n",
        "        for downloadable in selected_downloads:\n",
        "            url = 'https://maayanlab.cloud/static/hdfs/harmonizome/data/%s/%s' %\\\n",
        "                  (path, downloadable)\n",
        "            response = requests.get(url, stream=True)\n",
        "            filename = '%s/%s' % (dataset, downloadable)\n",
        "\n",
        "            # Not every dataset has all downloadables.\n",
        "            if response.status_code != 200:\n",
        "                continue\n",
        "\n",
        "            if decompress and 'txt.gz' in filename:\n",
        "                _download_and_decompress_file(response, filename)\n",
        "            else:\n",
        "                _download_file(response, filename)\n",
        "\n",
        "        print('%s downloaded.' % dataset)\n",
        "\n",
        "\n",
        "if __name__ == '__main__':\n",
        "    # Uncomment a dataset to download it.\n",
        "    download_datasets([\n",
        "        # ('Achilles Cell Line Gene Essentiality Profiles', 'achilles'),\n",
        "        # ('Allen Brain Atlas Adult Human Brain Tissue Gene Expression Profiles', 'brainatlasadulthuman'),\n",
        "        # ('Allen Brain Atlas Adult Mouse Brain Tissue Gene Expression Profiles', 'brainatlasadultmouse'),\n",
        "        # ('Allen Brain Atlas Developing Human Brain Tissue Gene Expression Profiles by Microarray', 'brainatlasdevelopmentalhumanmicroarray'),\n",
        "        # ('Allen Brain Atlas Developing Human Brain Tissue Gene Expression Profiles by RNA-seq', 'brainatlasdevelopmentalhumanrnaseq'),\n",
        "        # ('Allen Brain Atlas Prenatal Human Brain Tissue Gene Expression Profiles', 'brainatlasprenatalhuman'),\n",
        "        # ('BIND Biomolecular Interactions', 'bind'),\n",
        "        # ('BioGPS Cell Line Gene Expression Profiles', 'biogpsnci60'),\n",
        "        # ('BioGPS Human Cell Type and Tissue Gene Expression Profiles', 'biogpshuman'),\n",
        "        # ('BioGPS Mouse Cell Type and Tissue Gene Expression Profiles', 'biogpsmouse'),\n",
        "        # ('BioGRID Protein-Protein Interactions', 'biogrid'),\n",
        "        # ('Biocarta Pathways', 'biocarta'),\n",
        "        # ('CCLE Cell Line Gene CNV Profiles', 'cclecnv'),\n",
        "        # ('CCLE Cell Line Gene Expression Profiles', 'cclemrna'),\n",
        "        # ('CCLE Cell Line Gene Mutation Profiles', 'cclemut'),\n",
        "        # ('CHEA Transcription Factor Binding Site Profiles', 'chea'),\n",
        "        # ('CHEA Transcription Factor Targets', 'cheappi'),\n",
        "        # ('CMAP Signatures of Differentially Expressed Genes for Small Molecules', 'cmap'),\n",
        "        # ('COMPARTMENTS Curated Protein Localization Evidence Scores', 'jensencompartmentcurated'),\n",
        "        # ('COMPARTMENTS Experimental Protein Localization Evidence Scores', 'jensencompartmentexpts'),\n",
        "        # ('COMPARTMENTS Text-mining Protein Localization Evidence Scores', 'jensencompartmenttextmining'),\n",
        "        # ('CORUM Protein Complexes', 'corum'),\n",
        "        # ('COSMIC Cell Line Gene CNV Profiles', 'cosmiccnv'),\n",
        "        # ('COSMIC Cell Line Gene Mutation Profiles', 'cosmicmut'),\n",
        "        # ('CTD Gene-Chemical Interactions', 'ctdchemical'),\n",
        "        # ('CTD Gene-Disease Associations', 'ctddisease'),\n",
        "        # ('ClinVar SNP-Phenotype Associations', 'clinvar'),\n",
        "        # ('Combined Pathways Pathways', 'combinedpathways'),\n",
        "        # ('dbGAP Gene-Trait Associations', 'dbgap'),\n",
        "        # ('DEPOD Substrates of Phosphatases', 'depod'),\n",
        "        # ('DIP Protein-Protein Interactions', 'dip'),\n",
        "        # ('DISEASES Curated Gene-Disease Assocation Evidence Scores', 'jensendiseasecurated'),\n",
        "        # ('DISEASES Experimental Gene-Disease Assocation Evidence Scores', 'jensendiseaseexpts'),\n",
        "        # ('DISEASES Text-mining Gene-Disease Assocation Evidence Scores', 'jensendiseasetextmining'),\n",
        "        # ('DrugBank Drug Targets', 'drugbank'),\n",
        "        # ('ENCODE Histone Modification Site Profiles', 'encodehm'),\n",
        "        # ('ENCODE Transcription Factor Binding Site Profiles', 'encodetf'),\n",
        "        # ('ENCODE Transcription Factor Targets', 'encodetfppi'),\n",
        "        # ('ESCAPE Omics Signatures of Genes and Proteins for Stem Cells', 'escape'),\n",
        "        # ('GAD Gene-Disease Associations', 'gad'),\n",
        "        # ('GAD High Level Gene-Disease Associations', 'gadhighlevel'),\n",
        "        # ('GDSC Cell Line Gene Expression Profiles', 'gdsc'),\n",
        "        # ('GEO Signatures of Differentially Expressed Genes for Diseases', 'geodisease'),\n",
        "        # ('GEO Signatures of Differentially Expressed Genes for Gene Perturbations', 'geogene'),\n",
        "        # ('GEO Signatures of Differentially Expressed Genes for Kinase Perturbations', 'geokinase'),\n",
        "        # ('GEO Signatures of Differentially Expressed Genes for Small Molecules', 'geochemical'),\n",
        "        # ('GEO Signatures of Differentially Expressed Genes for Transcription Factor Perturbations', 'geotf'),\n",
        "        # ('GEO Signatures of Differentially Expressed Genes for Viral Infections', 'geovirus'),\n",
        "        # ('GO Biological Process Annotations', 'gobp'),\n",
        "        # ('GO Cellular Component Annotations', 'gocc'),\n",
        "        # ('GO Molecular Function Annotations', 'gomf'),\n",
        "        # ('GTEx Tissue Gene Expression Profiles', 'gtextissue'),\n",
        "        # ('GTEx Tissue Sample Gene Expression Profiles', 'gtexsample'),\n",
        "        # ('GTEx eQTL', 'gtexeqtl'),\n",
        "        # ('GWAS Catalog SNP-Phenotype Associations', 'gwascatalog'),\n",
        "        # ('GWASdb SNP-Disease Associations', 'gwasdbdisease'),\n",
        "        # ('GWASdb SNP-Phenotype Associations', 'gwasdbphenotype'),\n",
        "        # ('GeneRIF Biological Term Annotations', 'generif'),\n",
        "        # ('GeneSigDB Published Gene Signatures', 'genesigdb'),\n",
        "        # ('Graph of Medicine EHR Text-mining Clinical Term Annotations', 'graphofmedicine'),\n",
        "        # ('Guide to Pharmacology Chemical Ligands of Receptors', 'guidetopharmchemical'),\n",
        "        # ('Guide to Pharmacology Protein Ligands of Receptors', 'guidetopharmprotein'),\n",
        "        # ('HMDB Metabolites of Enzymes', 'hmdb'),\n",
        "        # ('HPA Cell Line Gene Expression Profiles', 'hpacelllines'),\n",
        "        # ('HPA Tissue Gene Expression Profiles', 'hpatissuesmrna'),\n",
        "        # ('HPA Tissue Protein Expression Profiles', 'hpatissuesprotein'),\n",
        "        # ('HPA Tissue Sample Gene Expression Profiles', 'hpasamples'),\n",
        "        # ('HPM Cell Type and Tissue Protein Expression Profiles', 'hpm'),\n",
        "        # ('HPO Gene-Disease Associations', 'hpo'),\n",
        "        # ('HPRD Protein-Protein Interactions', 'hprd'),\n",
        "        # ('Heiser et al., PNAS, 2011 Cell Line Gene Expression Profiles', 'heiser'),\n",
        "        # ('HuGE Navigator Gene-Phenotype Associations', 'hugenavigator'),\n",
        "        # ('Hub Proteins Protein-Protein Interactions', 'hubs'),\n",
        "        # ('HumanCyc Biomolecular Interactions', 'humancycppi'),\n",
        "        # ('HumanCyc Pathways', 'humancyc'),\n",
        "        # ('IntAct Biomolecular Interactions', 'intact'),\n",
        "        # ('InterPro Predicted Protein Domain Annotations', 'interpro'),\n",
        "        # ('JASPAR Predicted Transcription Factor Targets', 'jasparpwm'),\n",
        "        # ('KEA Substrates of Kinases', 'kea'),\n",
        "        # ('KEGG Biomolecular Interactions', 'keggppi'),\n",
        "        # ('KEGG Pathways', 'kegg'),\n",
        "        # ('Kinativ Kinase Inhibitor Bioactivity Profiles', 'kinativ'),\n",
        "        # ('KinomeScan Kinase Inhibitor Targets', 'kinomescan'),\n",
        "        # ('Klijn et al., Nat. Biotechnol., 2015 Cell Line Gene CNV Profiles', 'klijncnv'),\n",
        "        # ('Klijn et al., Nat. Biotechnol., 2015 Cell Line Gene Expression Profiles', 'klijnmrna'),\n",
        "        # ('Klijn et al., Nat. Biotechnol., 2015 Cell Line Gene Mutation Profiles', 'klijnmut'),\n",
        "        # ('LINCS L1000 CMAP Signatures of Differentially Expressed Genes for Gene Knockdowns', 'lincscmapgene'),\n",
        "        # ('LINCS L1000 CMAP Signatures of Differentially Expressed Genes for Small Molecules', 'lincscmapchemical'),\n",
        "        # ('LOCATE Curated Protein Localization Annotations', 'locate'),\n",
        "        # ('LOCATE Predicted Protein Localization Annotations', 'locatepredicted'),\n",
        "        # ('MPO Gene-Phenotype Associations', 'mgimpo'),\n",
        "        # ('MSigDB Cancer Gene Co-expression Modules', 'msigdbcomp'),\n",
        "        # ('MSigDB Signatures of Differentially Expressed Genes for Cancer Gene Perturbations', 'msigdbonc'),\n",
        "        # ('MiRTarBase microRNA Targets', 'mirtarbase'),\n",
        "        # ('MotifMap Predicted Transcription Factor Targets', 'motifmap'),\n",
        "        # ('NURSA Protein Complexes', 'nursa'),\n",
        "        # ('NURSA Protein-Protein Interactions', 'nursappi'),\n",
        "        # ('OMIM Gene-Disease Associations', 'omim'),\n",
        "        # ('PANTHER Biomolecular Interactions', 'pantherppi'),\n",
        "        # ('PANTHER Pathways', 'panther'),\n",
        "        # ('PID Biomolecular Interactions', 'pidppi'),\n",
        "        # ('PID Pathways', 'pid'),\n",
        "        # ('Pathway Commons Protein-Protein Interactions', 'pc'),\n",
        "        # ('PhosphoSitePlus Phosphosite-Disease Associations', 'phosphositeplusdisease'),\n",
        "        # ('PhosphoSitePlus Substrates of Kinases', 'phosphositeplus'),\n",
        "        # ('Phosphosite Textmining Biological Term Annotations', 'phosphositetextmining'),\n",
        "        # ('ProteomicsDB Cell Type and Tissue Protein Expression Profiles', 'proteomicsdb'),\n",
        "        # ('Reactome Biomolecular Interactions', 'reactomeppi'),\n",
        "        # ('Reactome Pathways', 'reactome'),\n",
        "        # ('Recon X Predicted Biomolecular Interactions', 'reconx'),\n",
        "        # ('Roadmap Epigenomics Cell and Tissue DNA Accessibility Profiles', 'epigenomicsdnaaccessibility'),\n",
        "        # ('Roadmap Epigenomics Cell and Tissue DNA Methylation Profiles', 'epigenomicsdnamethylation'),\n",
        "        # ('Roadmap Epigenomics Cell and Tissue Gene Expression Profiles', 'epigenomicsmrna'),\n",
        "        # ('Roadmap Epigenomics Histone Modification Site Profiles', 'epigenomicshm'),\n",
        "        # ('SILAC Phosphoproteomics Signatures of Differentially Phosphorylated Proteins for Drugs', 'silacdrug'),\n",
        "        # ('SILAC Phosphoproteomics Signatures of Differentially Phosphorylated Proteins for Gene Perturbations', 'silacgene'),\n",
        "        # ('SILAC Phosphoproteomics Signatures of Differentially Phosphorylated Proteins for Protein Ligands', 'silacligand'),\n",
        "        # ('SNPedia SNP-Phenotype Associations', 'snpedia'),\n",
        "        # ('TCGA Signatures of Differentially Expressed Genes for Tumors', 'tcga'),\n",
        "        # ('TISSUES Curated Tissue Protein Expression Evidence Scores', 'jensentissuecurated'),\n",
        "        # ('TISSUES Experimental Tissue Protein Expression Evidence Scores', 'jensentissueexpts'),\n",
        "        # ('TISSUES Text-mining Tissue Protein Expression Evidence Scores', 'jensentissuetextmining'),\n",
        "        # ('TRANSFAC Curated Transcription Factor Targets', 'transfac'),\n",
        "        # ('TRANSFAC Predicted Transcription Factor Targets', 'transfacpwm'),\n",
        "        # ('TargetScan Predicted Conserved microRNA Targets', 'targetscan'),\n",
        "        # ('TargetScan Predicted Nonconserved microRNA Targets', 'targetscannonconserved'),\n",
        "        # ('Virus MINT Protein-Viral Protein Interactions', 'virusmintppi'),\n",
        "        # ('Virus MINT Protein-Virus Interactions', 'virusmint'),\n",
        "        # ('Wikipathways Pathways', 'wikipathways'),\n",
        "    ], [\n",
        "         'gene_attribute_matrix.txt.gz',\n",
        "         'gene_attribute_edges.txt.gz',\n",
        "         'gene_set_library_crisp.txt.gz',\n",
        "         'gene_set_library_up_crisp.txt.gz',\n",
        "         'gene_set_library_dn_crisp.txt.gz',\n",
        "         'attribute_set_library_crisp.txt.gz',\n",
        "         'attribute_set_library_up_crisp.txt.gz',\n",
        "         'attribute_set_library_dn_crisp.txt.gz',\n",
        "         'gene_similarity_matrix_cosine.txt.gz',\n",
        "         'attribute_similarity_matrix_cosine.txt.gz',\n",
        "         'gene_list_terms.txt.gz',\n",
        "         'attribute_list_entries.txt.gz',\n",
        "         'processing_script.m'\n",
        "    ])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JQwpI-Pbe715",
        "outputId": "63bfaf55-b976-46c0-8c4f-c0794f67eb46"
      },
      "execution_count": 15,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Achilles Cell Line Gene Essentiality Profiles downloaded.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# aging_datasets_downloader.py\n",
        "\n",
        "# pip install anndata\n",
        "\n",
        "import os\n",
        "import requests\n",
        "from pathlib import Path\n",
        "import pandas as pd\n",
        "import gzip\n",
        "import zipfile\n",
        "import anndata\n",
        "\n",
        "# Diretório raiz para salvar todos os datasets\n",
        "BASE_DIR = Path(\"datasets\")\n",
        "BASE_DIR.mkdir(exist_ok=True)\n",
        "\n",
        "# Função auxiliar para baixar arquivos\n",
        "def download_file(name, url, subdir=\"general\"):\n",
        "    folder = BASE_DIR / subdir\n",
        "    folder.mkdir(parents=True, exist_ok=True)\n",
        "    filename = url.split(\"/\")[-1]\n",
        "    path = folder / filename\n",
        "\n",
        "    try:\n",
        "        r = requests.get(url)\n",
        "        r.raise_for_status()\n",
        "        with open(path, \"wb\") as f:\n",
        "            f.write(r.content)\n",
        "        print(f\"[OK] {name} salvo em {path}\")\n",
        "    except Exception as e:\n",
        "        print(f\"[ERRO] {name}: {e}\")\n",
        "\n",
        "    return str(path)\n",
        "\n",
        "# Função para resumir datasets\n",
        "\n",
        "def summarize_dataset(filepath):\n",
        "    try:\n",
        "        if filepath.endswith(\".xlsx\"):\n",
        "            df = pd.read_excel(filepath)\n",
        "        elif filepath.endswith(\".csv\"):\n",
        "            df = pd.read_csv(filepath)\n",
        "        elif filepath.endswith(\".txt.gz\"):\n",
        "            with gzip.open(filepath, 'rt') as f:\n",
        "                df = pd.read_csv(f, sep='\\t')\n",
        "        elif filepath.endswith(\".zip\"):\n",
        "            with zipfile.ZipFile(filepath, 'r') as zip_ref:\n",
        "                extract_path = Path(filepath).parent / (Path(filepath).stem + \"_extracted\")\n",
        "                zip_ref.extractall(extract_path)\n",
        "                print(f\"[INFO] Arquivos extraídos para {extract_path}\")\n",
        "                return None\n",
        "        elif filepath.endswith(\".h5ad\"):\n",
        "            adata = anndata.read_h5ad(filepath)\n",
        "            print(f\"[INFO] {filepath} | AnnData: {adata.shape} | Obs: {adata.obs.shape} | Vars: {adata.var.shape}\")\n",
        "            return adata\n",
        "        else:\n",
        "            print(f\"[WARN] Formato não suportado: {filepath}\")\n",
        "            return None\n",
        "\n",
        "        print(f\"[INFO] {filepath} | shape: {df.shape} | colunas: {list(df.columns)[:5]}...\")\n",
        "        return df\n",
        "    except Exception as e:\n",
        "        print(f\"[ERRO ao ler] {filepath}: {e}\")\n",
        "        return None\n",
        "\n",
        "# -----------------------------\n",
        "# MULTI-OMICS / MULTI-TISSUES\n",
        "# -----------------------------\n",
        "\n",
        "# Aging Atlas\n",
        "aging_atlas_files = {\n",
        "    \"Transcriptome Bulk\": \"https://ngdc.cncb.ac.cn/aging/download/data/Aging_Atlas_Bulk_Transcriptome.xlsx\",\n",
        "    \"Single-cell Transcriptome\": \"https://ngdc.cncb.ac.cn/aging/download/data/Aging_Atlas_Single_Cell.xlsx\",\n",
        "    \"Proteome\": \"https://ngdc.cncb.ac.cn/aging/download/data/Aging_Atlas_Proteome.xlsx\",\n",
        "    \"Pharmacogenomics\": \"https://ngdc.cncb.ac.cn/aging/download/data/Aging_Atlas_Pharmacogenomics.xlsx\"\n",
        "}\n",
        "\n",
        "for name, url in aging_atlas_files.items():\n",
        "    path = download_file(name, url, subdir=\"aging_atlas\")\n",
        "    summarize_dataset(path)\n",
        "\n",
        "# GTEx\n",
        "print(\"[INFO] GTEx requires registration and must be downloaded manually: https://gtexportal.org\")\n",
        "\n",
        "# Tabula Muris\n",
        "path = download_file(\"Tabula Muris Senis Raw Data\", \"https://figshare.com/ndownloader/files/22921744\", subdir=\"tabula_muris\")\n",
        "summarize_dataset(path)\n",
        "\n",
        "# Tabula Sapiens\n",
        "print(\"[INFO] Tabula Sapiens: acesse https://tabula-sapiens-portal.ds.czbiohub.org para download personalizado.\")\n",
        "\n",
        "# ENCODE\n",
        "print(\"[INFO] ENCODE datasets devem ser baixados via interface: https://www.encodeproject.org\")\n",
        "\n",
        "# Human Cell Atlas\n",
        "print(\"[INFO] HCA oferece API e interface: https://data.humancellatlas.org\")\n",
        "\n",
        "# Roadmap Epigenomics\n",
        "print(\"[INFO] Roadmap Epigenomics: baixe por https://egg2.wustl.edu/roadmap/web_portal/\")\n",
        "\n",
        "# AFCA\n",
        "path = download_file(\"Aging Fly Cell Atlas (AFCA)\", \"https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-022-05597-y/MediaObjects/41586_2022_5597_MOESM3_ESM.zip\", subdir=\"afca\")\n",
        "summarize_dataset(path)\n",
        "\n",
        "# Single Cell Portal\n",
        "print(\"[INFO] Broad Institute Single Cell Portal: https://singlecell.broadinstitute.org/single_cell\")\n",
        "\n",
        "# -----------------------------\n",
        "# LONGITUDINAL / AGE CLOCKS\n",
        "# -----------------------------\n",
        "\n",
        "# ClockBase\n",
        "print(\"[INFO] ClockBase requer acesso via https://clockbase.org\")\n",
        "\n",
        "# ROSMAP\n",
        "print(\"[INFO] ROSMAP disponível via Synapse: https://www.synapse.org/#!Synapse:syn3219045\")\n",
        "\n",
        "# GEO (exemplo GSE201338)\n",
        "path = download_file(\"GSE201338 metadata\", \"https://ftp.ncbi.nlm.nih.gov/geo/series/GSE201nnn/GSE201338/matrix/GSE201338_series_matrix.txt.gz\", subdir=\"geo\")\n",
        "summarize_dataset(path)\n",
        "\n",
        "# Mammalian Aging Cell Atlas (MACA)\n",
        "path = download_file(\"MACA (Mouse Aging Single-Cell)\", \"https://figshare.com/ndownloader/files/33835075\", subdir=\"maca\")\n",
        "summarize_dataset(path)\n",
        "\n",
        "# SenNet\n",
        "print(\"[INFO] SenNet data hub: https://sennetconsortium.org\")\n",
        "\n",
        "# Multi-Omics Longitudinal Fibroblasts\n",
        "print(\"[INFO] Download pelo repositório suplementar do estudo: https://www.nature.com/articles/s41586-020-2326-1\")\n",
        "\n",
        "# BrainSpan / Allen Brain Atlas\n",
        "print(\"[INFO] BrainSpan download: https://www.brainspan.org/static/download.html\")\n",
        "\n",
        "# LifeTime Initiative\n",
        "print(\"[INFO] LifeTime resources: https://lifetime-fetflagship.eu\")\n",
        "\n",
        "# Framingham Heart Study\n",
        "print(\"[INFO] FHS data via dbGaP: https://www.nhlbi.nih.gov/science/framingham-heart-study\")\n",
        "\n",
        "# -----------------------------\n",
        "# REJUVENATION / INTERVENTIONS\n",
        "# -----------------------------\n",
        "\n",
        "# SINGULAR (Cell Rejuvenation Atlas)\n",
        "print(\"[INFO] Acesse o Cell Rejuvenation Atlas (SINGULAR): https://rejuvenome.org\")\n",
        "\n",
        "# TPE-IVIG Study\n",
        "print(\"[INFO] Dataset suplementar via artigo original: https://www.nature.com/articles/s41467-023-38028-0\")\n",
        "\n",
        "# Rejuvenation Roadmap\n",
        "print(\"[INFO] Rejuvenation Roadmap datasets: https://www.lifespan.io/road-maps/the-rejuvenation-roadmap/\")\n",
        "\n",
        "# iPSC datasets\n",
        "print(\"[INFO] Acesse repositórios iPSC via GEO ou https://stemcellcommons.org\")\n",
        "\n",
        "# DGIdb\n",
        "print(\"[INFO] DGIdb API e downloads: https://www.dgidb.org\")\n",
        "\n",
        "# DrugBank\n",
        "print(\"[INFO] DrugBank requer registro para download: https://go.drugbank.com/releases/latest\")\n",
        "\n",
        "# -----------------------------\n",
        "# EXTRA DATABASES ADDED\n",
        "# -----------------------------\n",
        "\n",
        "# agingbiotech.info\n",
        "print(\"[INFO] agingbiotech.info (empresas e intervenções): https://agingbiotech.info\")\n",
        "\n",
        "# OpenGenes\n",
        "print(\"[INFO] OpenGenes database: https://open-genes.org\")\n",
        "\n",
        "# Biomarkers of Aging Consortium\n",
        "print(\"[INFO] Biomarkers of Aging Consortium (data access upon request): https://www.biomarkers consortium.org\")\n",
        "\n",
        "# CZI Cellxgene Discover (Census)\n",
        "print(\"[INFO] CZI cellxgene Discover portal: https://census.cellxgene.cziscience.com\")\n",
        "\n",
        "# gnomAD\n",
        "print(\"[INFO] gnomAD data (genome/variant frequency): https://gnomad.broadinstitute.org/downloads\")\n",
        "\n",
        "# -----------------------------\n",
        "# Finalização\n",
        "# -----------------------------\n",
        "print(\"\\n[✔] Script finalizado. Datasets disponíveis foram baixados e resumidos.\")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Iw2ciENpkMto",
        "outputId": "a0b4782e-ddfd-403f-b63c-706021ecea0a"
      },
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[ERROR] Transcriptome Bulk: 404 Client Error:  for url: https://ngdc.cncb.ac.cn/aging/download/data/Aging_Atlas_Bulk_Transcriptome.xlsx\n",
            "[ERROR] Single-cell Transcriptome: 404 Client Error:  for url: https://ngdc.cncb.ac.cn/aging/download/data/Aging_Atlas_Single_Cell.xlsx\n",
            "[ERROR] Proteome: 404 Client Error:  for url: https://ngdc.cncb.ac.cn/aging/download/data/Aging_Atlas_Proteome.xlsx\n",
            "[ERROR] Pharmacogenomics: 404 Client Error:  for url: https://ngdc.cncb.ac.cn/aging/download/data/Aging_Atlas_Pharmacogenomics.xlsx\n",
            "[INFO] GTEx requires registration and must be downloaded manually: https://gtexportal.org\n",
            "[ERROR] Tabula Muris Senis Raw Data: 404 Client Error: Not Found for url: https://figshare.com/ndownloader/files/22921744\n",
            "[INFO] Tabula Sapiens: access https://tabula-sapiens-portal.ds.czbiohub.org for personalized download.\n",
            "[INFO] ENCODE datasets must be downloaded via the interface: https://www.encodeproject.org\n",
            "[INFO] HCA offers API and interface: https://data.humancellatlas.org\n",
            "[INFO] Roadmap Epigenomics: download by https://egg2.wustl.edu/roadmap/web_portal/\n",
            "[ERROR] Aging Fly Cell Atlas (AFCA): 403 Client Error: Forbidden for url: https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-022-05597-y/MediaObjects/41586_2022_5597_MOESM3_ESM.zip\n",
            "[INFO] Broad Institute Single Cell Portal: https://singlecell.broadinstitute.org/single_cell\n",
            "[INFO] OpenGenes database: https://open-genes.org\n",
            "[INFO] CZI cellxgene Discover portal: https://census.cellxgene.cziscience.com\n",
            "[INFO] gnomAD data (genome/variant frequency): https://gnomad.broadinstitute.org/downloads\n",
            "[INFO] ClockBase requires access via https://clockbase.org\n",
            "[INFO] ROSMAP available via Synapse: https://www.synapse.org/#!Synapse:syn3219045\n",
            "[OK] GSE201338 metadata saved in datasets/geo/GSE201338_series_matrix.txt.gz\n",
            "[ERROR] MACA (Mouse Aging Single-Cell): 404 Client Error: Not Found for url: https://figshare.com/ndownloader/files/33835075\n",
            "[INFO] SenNet data hub: https://sennetconsortium.org\n",
            "[INFO] Download from the study's supplementary repository: https://www.nature.com/articles/s41586-020-2326-1\n",
            "[INFO] BrainSpan download: https://www.brainspan.org/static/download.html\n",
            "[INFO] LifeTime resources: https://lifetime-fetflagship.eu\n",
            "[INFO] FHS data via dbGaP: https://www.nhlbi.nih.gov/science/framingham-heart-study\n",
            "[INFO] Access the Cell Rejuvenation Atlas (SINGULAR): https://rejuvenome.org\n",
            "[INFO] Supplementary dataset via original article: https://www.nature.com/articles/s41467-023-38028-0\n",
            "[INFO] Rejuvenation Roadmap datasets: https://www.lifespan.io/road-maps/the-rejuvenation-roadmap/\n",
            "[INFO] Access iPSC repositories via GEO or https://stemcellcommons.org\n",
            "[INFO] DGIdb API and downloads: https://www.dgidb.org\n",
            "[INFO] DrugBank requires registration to download: https://go.drugbank.com/releases/latest\n",
            "\n",
            "[✔] Script finished. Available datasets have been downloaded. See directories at ./datasets/\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# PARSING DATA"
      ],
      "metadata": {
        "id": "A_v7rrKZqsKj"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import gzip\n",
        "\n",
        "def read_gz_file(filepath):\n",
        "    try:\n",
        "        with gzip.open(filepath, 'rt') as f:\n",
        "            for line in f:\n",
        "                print(line, end='') # process each line as needed\n",
        "    except FileNotFoundError:\n",
        "        print(f\"Error: File not found at {filepath}\")\n",
        "    except Exception as e:\n",
        "        print(f\"An error occurred: {e}\")\n",
        "\n",
        "read_gz_file('/content/datasets/geo/GSE201338_series_matrix.txt.gz')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "SKiA8_3WoQJF",
        "outputId": "04a170c8-d29c-4f71-ca80-6aad20385ae8"
      },
      "execution_count": 19,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "!Series_title\t\"Combining Stem Cell Rejuvenation and Senescence Targeting to Synergistically Extend Lifespan\"\n",
            "!Series_geo_accession\t\"GSE201338\"\n",
            "!Series_status\t\"Public on Apr 24 2022\"\n",
            "!Series_submission_date\t\"Apr 22 2022\"\n",
            "!Series_last_update_date\t\"Sep 21 2022\"\n",
            "!Series_summary\t\"We combine transient stem cell rejuvenation with targeted removal of senescent cells to test the hypothesis that simultaneously targeting both cell-fate based aging mechanisms will maximize life and health span benefits.\"\n",
            "!Series_overall_design\t\"Drosophila midgut RNA profiles of tubGal80ts > UAS-TdTomato (WT), armGal4; tubGal80ts > UAS-OKSM (OKSM), armGal4; tubGal80ts > UAS-Sen (Sen) and armGal4; tubGal80ts > UAS-Sen; UAS-OKSM (Sen_OKSM); tubGal80ts > UAS-TdTomato (WT), esgGal4; tubGal80ts > UAS-OKSM (OKSM), esgGal4; tubGal80ts > UAS-Sen (Sen) and esgGal4; tubGal80ts > UAS-Sen; UAS-OKSM (Sen_OKSM)\"\n",
            "!Series_overall_design\t\"\"\n",
            "!Series_overall_design\t\"Updates: [Sept. 20, 2022] The GSM6058923 and GSM6058926 metadata were swapped in accordance with the sample identities. The associated data files did not change but were renamed accordingly.\"\n",
            "!Series_type\t\"Expression profiling by high throughput sequencing\"\n",
            "!Series_contributor\t\"Nicholas,S,Tolwinski\"\n",
            "!Series_contributor\t\"Prameet,,Kaur\"\n",
            "!Series_contributor\t\"Agimaa,,Otgonbaatar\"\n",
            "!Series_contributor\t\"Anupriya,,Ramamoorthy\"\n",
            "!Series_contributor\t\"Ellora,,Chua\"\n",
            "!Series_contributor\t\"Nathan,P,Harmston\"\n",
            "!Series_contributor\t\"Jan,,Gruber\"\n",
            "!Series_sample_id\t\"GSM6058907 GSM6058908 GSM6058909 GSM6058910 GSM6058911 GSM6058912 GSM6058913 GSM6058914 GSM6058915 GSM6058916 GSM6058917 GSM6058918 GSM6058919 GSM6058920 GSM6058921 GSM6058922 GSM6058923 GSM6058924 GSM6058925 GSM6058926 GSM6058927 GSM6058928 GSM6058929 \"\n",
            "!Series_contact_name\t\"Nicholas,,Tolwinski\"\n",
            "!Series_contact_email\t\"nicholas.tolwinski@yale-nus.edu.sg\"\n",
            "!Series_contact_institute\t\"Yale-NUS College\"\n",
            "!Series_contact_address\t\"16 #01-220, College Ave West\"\n",
            "!Series_contact_city\t\"Singapore\"\n",
            "!Series_contact_zip/postal_code\t\"138527\"\n",
            "!Series_contact_country\t\"Singapore\"\n",
            "!Series_supplementary_file\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE201nnn/GSE201338/suppl/GSE201338_RAW.tar\"\n",
            "!Series_platform_id\t\"GPL21306\"\n",
            "!Series_platform_taxid\t\"7227\"\n",
            "!Series_sample_taxid\t\"7227\"\n",
            "!Series_relation\t\"BioProject: https://www.ncbi.nlm.nih.gov/bioproject/PRJNA830891\"\n",
            "\n",
            "!Sample_title\t\"Midgut with OKSM overexpression Rep 1\"\t\"Midgut with OKSM overexpression Rep 2\"\t\"Midgut with OKSM overexpression Rep 3\"\t\"Midgut with senolytic overepxression Rep 2\"\t\"Midgut with senolytic overepxression Rep 3\"\t\"Midgut WT (OKSM +Sen control) Rep 1\"\t\"Midgut WT (OKSM +Sen control) Rep 2\"\t\"Midgut WT (OKSM +Sen control) Rep 3\"\t\"Midgut with OKSM & senolytic overexpression Rep 1\"\t\"Midgut with OKSM & senolytic overexpression Rep 2\"\t\"Midgut with OKSM & senolytic overexpression Rep 3\"\t\"Intestinal stem cell with OKSM overexpression Rep 1\"\t\"Intestinal stem cell with OKSM overexpression Rep 2\"\t\"Intestinal stem cell with OKSM overexpression Rep 3\"\t\"Intestinal stem cell with senolytic overepxression Rep 1\"\t\"Intestinal stem cell with senolytic overepxression Rep 2\"\t\"Intestinal stem cell with OKSM & senolytic overexpression Rep 3\"\t\"Intestinal stem cell with OKSM & senolytic overexpression Rep 1\"\t\"Intestinal stem cell with OKSM & senolytic overexpression Rep 2\"\t\"Intestinal stem cell with senolytic overepxression Rep 3\"\t\"Intestinal stem cell WT (OKSM +Sen control) Rep 1\"\t\"Intestinal stem cell WT (OKSM +Sen control) Rep 2\"\t\"Intestinal stem cell WT (OKSM +Sen control) Rep 3\"\n",
            "!Sample_geo_accession\t\"GSM6058907\"\t\"GSM6058908\"\t\"GSM6058909\"\t\"GSM6058910\"\t\"GSM6058911\"\t\"GSM6058912\"\t\"GSM6058913\"\t\"GSM6058914\"\t\"GSM6058915\"\t\"GSM6058916\"\t\"GSM6058917\"\t\"GSM6058918\"\t\"GSM6058919\"\t\"GSM6058920\"\t\"GSM6058921\"\t\"GSM6058922\"\t\"GSM6058923\"\t\"GSM6058924\"\t\"GSM6058925\"\t\"GSM6058926\"\t\"GSM6058927\"\t\"GSM6058928\"\t\"GSM6058929\"\n",
            "!Sample_status\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\t\"Public on Apr 24 2022\"\n",
            "!Sample_submission_date\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\t\"Apr 22 2022\"\n",
            "!Sample_last_update_date\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Sep 20 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Sep 20 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\t\"Apr 26 2022\"\n",
            "!Sample_type\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\t\"SRA\"\n",
            "!Sample_channel_count\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\t\"1\"\n",
            "!Sample_source_name_ch1\t\"Midgut\"\t\"Midgut\"\t\"Midgut\"\t\"Midgut\"\t\"Midgut\"\t\"Midgut\"\t\"Midgut\"\t\"Midgut\"\t\"Midgut\"\t\"Midgut\"\t\"Midgut\"\t\"Intenstinal Stem Cell\"\t\"Intenstinal Stem Cell\"\t\"Intenstinal Stem Cell\"\t\"Intenstinal Stem Cell\"\t\"Intenstinal Stem Cell\"\t\"Intestinal Stem Cell\"\t\"Intenstinal Stem Cell\"\t\"Intenstinal Stem Cell\"\t\"Intestinal Stem Cell\"\t\"Intenstinal Stem Cell\"\t\"Intenstinal Stem Cell\"\t\"Intenstinal Stem Cell\"\n",
            "!Sample_organism_ch1\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\t\"Drosophila melanogaster\"\n",
            "!Sample_characteristics_ch1\t\"tissue: midgut\"\t\"tissue: midgut\"\t\"tissue: midgut\"\t\"tissue: midgut\"\t\"tissue: midgut\"\t\"tissue: midgut\"\t\"tissue: midgut\"\t\"tissue: midgut\"\t\"tissue: midgut\"\t\"tissue: midgut\"\t\"tissue: midgut\"\t\"tissue: intestinal stem cells\"\t\"tissue: intestinal stem cells\"\t\"tissue: intestinal stem cells\"\t\"tissue: intestinal stem cells\"\t\"tissue: intestinal stem cells\"\t\"tissue: intestinal stem cells\"\t\"tissue: intestinal stem cells\"\t\"tissue: intestinal stem cells\"\t\"tissue: intestinal stem cells\"\t\"tissue: intestinal stem cells\"\t\"tissue: intestinal stem cells\"\t\"tissue: intestinal stem cells\"\n",
            "!Sample_characteristics_ch1\t\"genotype: OKSM\"\t\"genotype: OKSM\"\t\"genotype: OKSM\"\t\"genotype: Senolytic\"\t\"genotype: Senolytic\"\t\"genotype: Control\"\t\"genotype: Control\"\t\"genotype: Control\"\t\"genotype: Senolytic_OKSM\"\t\"genotype: Senolytic_OKSM\"\t\"genotype: Senolytic_OKSM\"\t\"genotype: OKSM\"\t\"genotype: OKSM\"\t\"genotype: OKSM\"\t\"genotype: Senolytic\"\t\"genotype: Senolytic\"\t\"genotype: Senolytic_OKSM\"\t\"genotype: Senolytic_OKSM\"\t\"genotype: Senolytic_OKSM\"\t\"genotype: Senolytic\"\t\"genotype: Control\"\t\"genotype: Control\"\t\"genotype: Control\"\n",
            "!Sample_treatment_protocol_ch1\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\t\"Transgenic flies were maintained under a white light lamp at all times.\"\n",
            "!Sample_growth_protocol_ch1\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\t\"Drosophila were maintained at standard humidity and temperature (25⁰C) with food containing 6g Bacto agar, 114g glucose, 56g cornmeal, 25g Brewer’s yeast and 20ml of 10% Nipagin in 1L final volume.\"\n",
            "!Sample_molecule_ch1\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\t\"total RNA\"\n",
            "!Sample_extract_protocol_ch1\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\t\"Adult fly midguts were dissected in 200 µl of 1x PBS in a PYREX™ Spot Plates concave glass dish (FisherScientific). The midguts were rinsed with PBS and stained with 10mg/ml Hoechst 33342 diluted 200 times in 1X PBS for 1min. Subsequently, the guts were carefully transferred onto a small droplet of 1x PBS on a 35mm glass bottom dish. Using fine forceps, the gut was repositioned to resemble its natural orientation. PBS was then removed from the area surrounding the gut, leaving a small amount of excess PBS to hold the gut in place and prevent desiccation. The 3mm glass bottom dish was then mounted onto the Zeiss LSM800 (Carl Zeiss AG, Germany) for imaging. For each construct, midguts from at least 3 flies were dissected and imaged at the 25% percentile from the anterior midgut.\"\n",
            "!Sample_extract_protocol_ch1\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\t\"The extracted RNA was quantified using Nanodrop (Thermo Fisher Scientific). Library preparation was performed using 1µg of total RNA and sequencing was performed using Illumina HiSeq 4000 System (2 × 151 bp read length, 40 million reads per sample) by NovogeneAIT Genomics (Singapore).\"\n",
            "!Sample_taxid_ch1\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\t\"7227\"\n",
            "!Sample_data_processing\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\t\"The raw fastq files were aligned to the dm6 genome (Ensembl v97) using STAR v2.7.1a\"\n",
            "!Sample_data_processing\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\t\"The reads were then quantified using RSEM v1.3.1\"\n",
            "!Sample_data_processing\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\t\"Assembly: dm6\"\n",
            "!Sample_data_processing\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\t\"Supplementary files format and content: genes.results were generated using RSEM v1.3.1 and includes the quantified reads for each sample\"\n",
            "!Sample_platform_id\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\t\"GPL21306\"\n",
            "!Sample_contact_name\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\t\"Nicholas,,Tolwinski\"\n",
            "!Sample_contact_email\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\t\"nicholas.tolwinski@yale-nus.edu.sg\"\n",
            "!Sample_contact_institute\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\t\"Yale-NUS College\"\n",
            "!Sample_contact_address\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\t\"16 #01-220, College Ave West\"\n",
            "!Sample_contact_city\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\n",
            "!Sample_contact_zip/postal_code\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\t\"138527\"\n",
            "!Sample_contact_country\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\t\"Singapore\"\n",
            "!Sample_data_row_count\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\t\"0\"\n",
            "!Sample_instrument_model\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\t\"Illumina HiSeq 4000\"\n",
            "!Sample_library_selection\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\t\"cDNA\"\n",
            "!Sample_library_source\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\t\"transcriptomic\"\n",
            "!Sample_library_strategy\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\t\"RNA-Seq\"\n",
            "!Sample_relation\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740565\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740564\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740563\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740562\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740561\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740560\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740559\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740558\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740557\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740556\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740555\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740554\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740553\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740552\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740551\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740550\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740549\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740548\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740547\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740546\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740545\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740544\"\t\"BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27740543\"\n",
            "!Sample_relation\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972143\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972144\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972145\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972146\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972147\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972148\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972149\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972150\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972151\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972152\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972153\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972154\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972155\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972156\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972157\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972158\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972159\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972160\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972161\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972162\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972163\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972164\"\t\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX14972165\"\n",
            "!Sample_supplementary_file_1\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058907/suppl/GSM6058907_AT_OKSM_1.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058908/suppl/GSM6058908_AT_OKSM_2.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058909/suppl/GSM6058909_AT_OKSM_3.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058910/suppl/GSM6058910_AT_Sen_2.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058911/suppl/GSM6058911_AT_Sen_3.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058912/suppl/GSM6058912_AT_TdTom_1.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058913/suppl/GSM6058913_AT_TdTom_2.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058914/suppl/GSM6058914_AT_TdTom_3.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058915/suppl/GSM6058915_AT_Sen_OKSM_1.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058916/suppl/GSM6058916_AT_Sen_OKSM_2.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058917/suppl/GSM6058917_AT_Sen_OKSM_3.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058918/suppl/GSM6058918_O1.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058919/suppl/GSM6058919_O2.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058920/suppl/GSM6058920_O3.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058921/suppl/GSM6058921_S1.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058922/suppl/GSM6058922_S2.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058923/suppl/GSM6058923_SO3.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058924/suppl/GSM6058924_SO1.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058925/suppl/GSM6058925_SO2.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058926/suppl/GSM6058926_S3.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058927/suppl/GSM6058927_Control_1.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058928/suppl/GSM6058928_Control_2.genes.results.gz\"\t\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6058nnn/GSM6058929/suppl/GSM6058929_Control_3.genes.results.gz\"\n",
            "!series_matrix_table_begin\n",
            "\"ID_REF\"\t\"GSM6058907\"\t\"GSM6058908\"\t\"GSM6058909\"\t\"GSM6058910\"\t\"GSM6058911\"\t\"GSM6058912\"\t\"GSM6058913\"\t\"GSM6058914\"\t\"GSM6058915\"\t\"GSM6058916\"\t\"GSM6058917\"\t\"GSM6058918\"\t\"GSM6058919\"\t\"GSM6058920\"\t\"GSM6058921\"\t\"GSM6058922\"\t\"GSM6058923\"\t\"GSM6058924\"\t\"GSM6058925\"\t\"GSM6058926\"\t\"GSM6058927\"\t\"GSM6058928\"\t\"GSM6058929\"\n",
            "!series_matrix_table_end\n"
          ]
        }
      ]
    }
  ]
}